packages/teams-js/src/public/meeting.ts

```diff
@@ -560,4 +594,93 @@ export namespace meeting {
       sendMessageToParent('meeting.appShareButton.setOptions', [shareInformation]);
     }
   }
+
+  /**
+   * Have the external application handle audio (mic & speaker) and turn off host audio.
+   * set isAppHandlingAudio=true in params, will set the Host(Teams) audio to go audioless
+   *   Register the mic mute status change events, and App will be receive the events.
+   * set isAppHandlingAudio=false in params, will restore the Host(Teams) audio
+   *   Deregister the mic mute status change events and App will not be receive the events.
+   *
+   * @throws standard parameter validation error
+   * @param requestAppAudioHandlingParams - RequestAppAudioHandlingParams interface, and contains
+   *   isAppHandlingAudio - true - expect App to handle the audio and Teams to go audioless.
+   *     false - Audio transferred to Teams and Teams to restore the audio.
+   *   callbackMicMuteStateChangedHandler - Callback contains mic status param.
+   * @param callback - Callback contains error and operation result parameters.
+   * error can either contain an error of type SdkError (error indication), or null (non-error indication)
+   * result can either contain a true boolean value (successful termination), or null (unsuccessful fetch)
+   * @beta
+   */
+  export function requestAppAudioHandling(
+    requestAppAudioHandlingParams: RequestAppAudioHandlingParams,
+    callback: (error: SdkError | null, isHostAudioless: boolean | null) => void,
+  ): void {
+    if (!callback) {
+      throw new Error('[requestAppAudioHandling] Callback response cannot be null');
+    }
+    if (!requestAppAudioHandlingParams.callbackMicMuteStateChangedHandler) {
+      throw new Error('[requestAppAudioHandling] Callback Mic mute state handler cannot be null');
+    }
+    ensureInitialized(runtime, FrameContexts.sidePanel, FrameContexts.meetingStage);
+
+    if (requestAppAudioHandlingParams.isAppHandlingAudio) {
+      startAppAudioHandling(requestAppAudioHandlingParams, callback);
+    } else {
+      stopAppAudioHandling(requestAppAudioHandlingParams, callback);
+    }
+  }
+
+  function startAppAudioHandling(
+    requestAppAudioHandlingParams: RequestAppAudioHandlingParams,
+    callback: (error: SdkError | null, isHostAudioless: boolean | null) => void,
+  ): void {
+    const callbackInternalRequest = (error: SdkError | null, isHostAudioless: boolean | null): void => {
+      if (error && isHostAudioless != null) {
+        throw new Error('[requestAppAudioHandling] Callback response - both parameters cannot be set');
```

Comment:
@TrevorJoelHarris Question about throwing errors here. Since this is in a `sendMessageToParent` callback, would the app still be able to catch these errors? If not, then it seems like using the optional error parameter in the callback is the only way to tell the app about these errors?

Reply:
No, the app can't really catch these errors, and in this case I think that's correct. These errors are truly exceptional: the host is passing the app malformed data in response to a request. I'd much rather have this function fail immediately so we get a bug opened against us if something goes wrong here. Good question!